### PR TITLE
*: support predeclared golangci-lint linter

### LINKED ISSUE #30733: Enable golangci-lint `predeclared`

### PR DESCRIPTION
**Pull request by wangggong**
Signed-off-by: wangggong <793160615@qq.com>

### What problem does this PR solve?

Issue Number: close #30733

Problem Summary:

### What is changed and how it works?

Support `predeclared` golangci-lint linter and a bunch of variable renaming.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by wangggong** [comment]
@tangenta PTAL

File: br/pkg/conn/conn.go [hunk 1]
```diff
@@ -90,11 +90,11 @@ func (p *Pool) Get(ctx context.Context) (*grpc.ClientConn, error) {
 }
 
 // NewConnPool creates a new Pool by the specified conn factory function and capacity.
-func NewConnPool(cap int, newConn func(ctx context.Context) (*grpc.ClientConn, error)) *Pool {
+func NewConnPool(c int, nc func(ctx context.Context) (*grpc.ClientConn, error)) *Pool {
```
Comment (hawkingrei) [inline br/pkg/conn/conn.go:93]: I think it use ```capacity``` better than ```c```.

Comment (wangggong) [reply br/pkg/conn/conn.go]: Got it. done

**Review by tangenta — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Comment by tangenta** [comment]
/hold

@wangggong Could you update for the following cases?

```
[2022-02-09T08:18:35.268Z] ddl/placement_sql_test.go:296:9: variable close has same name as predeclared identifier (predeclared)

[2022-02-09T08:18:35.268Z] 	store, close := testkit.CreateMockStore(t)

[2022-02-09T08:18:35.268Z] 	       ^

[2022-02-09T08:18:35.268Z] ddl/placement_sql_test.go:653:9: variable close has same name as predeclared identifier (predeclared)

[2022-02-09T08:18:35.268Z] 	store, close := testkit.CreateMockStore(t)
```

**Comment by tangenta** [comment]
/hold

@wangggong https://github.com/wangggong/tidb/pull/1 PTAL

**Comment by wangggong** [comment]
> /hold
> 
> @wangggong [wangggong#1](https://github.com/wangggong/tidb/pull/1) PTAL

LGTM, merged. Thanks...

@tangenta Still need an approving, so PTAL

**Comment by hawkingrei** [comment]
@XuHuaiyu @qw4990 @cfzjywxk PTAL

**Review by tangenta — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**